### PR TITLE
Fix playground timeout (fix #3216, #3509).

### LINKED ISSUE #3216: Response Timeout as a result of compilation big contracts.
## Area

[] *Plutus Foundation* Related to the GHC plugin, Haskell-to-Plutus compiler, on-chain code
[X] *Plutus Application Framework* Related to the Plutus application backend (PAB), emulator, Plutus libraries
[] *Marlowe* Related to Marlowe
[] *Other* Any other topic (Playgrounds, etc.)

## Summary

Timeout error as the compilation result for big contracts.

## Steps to reproduce

Press "Compile" in the simulator as usual.

## Expected behavior

It should wait at least for a `maxInterpretationTime`.

## System info (please complete the following information):

- OS: Ubuntu
- Version: 20.04
- Plutus version or commit hash: `47bee0d68`

## Screenshots and attachments

Screenshot from 2021-05-13 13-45-42
Screenshot from 2021-05-13 20-25-52

## Additional context

30 sec timeout.

### LINKED ISSUE #3509: ResponseTimeout While Evaluating Contracts in the Playground
## Area

[] *Plutus Foundation* Related to the GHC plugin, Haskell-to-Plutus compiler, on-chain code
[] *Plutus Application Framework* Related to the Plutus application backend (PAB), emulator, Plutus libraries
[] *Marlowe* Related to Marlowe
[X] *Other* Any other topic (Playgrounds, etc.)

## Summary

Evaluating contracts in the playground time-out after 30 seconds.  Leaving this error:

Interpreter Errors
ConnectionError (HttpExceptionRequest Request { host = "localhost" port = 8080 secure = False requestHeaders = [("Accept","application/json;charset=utf-8,application/json"),("Content-Type","application/json;charset=utf-8")] path = "/runghc" queryString = "" method = "POST" proxy = Nothing rawBody = False redirectCount = 10 responseTimeout = ResponseTimeoutDefault requestVersion = HTTP/1.1 } ResponseTimeout)
Please try again or contact support for assistance.

## Steps to reproduce

1. Start server and client inside nix-shell
2. Compile contract
3. Simulate
4. Evaluate contract

## Expected behavior

To simulate contract scenario.

## System info (please complete the following information):

- OS: Ubuntu
- Version 20.04
- ea0ca4e9f9821a9dbfc5255fa0f42b6f2b3887c4

## Screenshots and attachments

None

## Additional context

Issue was brought up in the Plutus Pioneer Q/A and it is believed that the time-out setpoint is hardcoded.  It would be nice to be able to set it manually if needed.

### PR DESCRIPTION
**Pull request by ak3n**
Add plutus-playground-server maxInterpretation timeout cli option. (fix #3216, fix #3509)

---

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-playground-server/app/Main.hs [hunk 1]
```diff
@@ -55,10 +56,16 @@ webserverCommandParser =
                 (short 'p' <> long "port" <> help "Webserver port number" <>
                  showDefault <>
                  value 8080)
+        _maxInterpretationTime <-
+            option
+                auto
+                (short 'i' <> long "interpretation" <> help "Max interpretation time" <>
```
Comment (j-mueller) [inline plutus-playground-server/app/Main.hs:62]: the help text should say the unit (seconds)

**Review by j-mueller — APPROVED** [review]

File: plutus-playground-server/app/Main.hs (hunk 1 above)
Comment (james-iohk) [reply plutus-playground-server/app/Main.hs]: Changed to  "Max interpretation time (seconds)"

**Comment by james-iohk** [comment]
We could merge this but bear in mind the port/timeout cli options appear to be ignored. E.g.
```
[nix-shell:~/dev/plutus/plutus-playground-client]$ plutus-playground-server -p 8081 -i 150
plutus-playground-server: for development use only
[Info] Running: (Nothing,Webserver {_port = 8080, _maxInterpretationTime = 80s})
```
If not fixed here please raise a new issue (or ask me to).

**Comment by luigy** [comment]
@james-iohk oh that's because what gets exposed in a nix-shell is wrapper script for `plutus-playground-server` that doesn't take any more args. lmk if you would like me to push here so that it does take them?

**Comment by james-iohk** [comment]
> lmk if you would like me to push here so that it does take them?

@luigy If you have a fix please do commit and I'll give it a go.

**Comment by luigy** [comment]
@james-iohk pushed

**Pull request merged**